### DISCUSSION TITLE
Documentation on how to create a resolver with query filtering

### DISCUSSION THREAD

**Discussion by jasonbiondo** [gatsbyjs/gatsby#28126] [Help]
Can someone explain how to create a resolver for something like limiting the amount of images returned in query.  Lets assume you have

node -> images -> src

and I want to do 

node -> images(limit: 2) -> src 

so only 2 images would be queryied via graphql.  How can we create these subfilters? I. assume it's with a resolver but I see no example of this in the docs.

**Comment by pvdz** [comment]
Hi @jasonbiondo . I'm going to convert your issue to a discussion :)

**Reply by jasonbiondo** [reply]
@pvdz Thank you but does anyone know how to do this?

**Comment by vladar** [comment]
You can do this with custom resolvers using `createResolvers` API. See those links for some examples:

https://www.gatsbyjs.com/docs/node-apis/#createResolvers
https://github.com/gatsbyjs/gatsby/blob/master/examples/using-type-definitions/gatsby-node.js

Also in blog posts:
https://www.gatsbyjs.com/blog/2019-03-04-new-schema-customization/
https://www.gatsbyjs.com/blog/2019-03-18-releasing-new-schema-customization/

Also given your description, something like this will probably work:

```js
// in gatsby-node
exports.createResolvers = ({ createResolvers }) => {
  createResolvers({
    NodeType: {
      images: {
        args: { limit: `Int` },
        resolve: async (root, args, context, info) => {
          const allImages = await info.originalResolver(root, args, context, info) || []
          return args.limit ? allImages.slice(0, args.limit) : allImages
        }
      }
    }
  })
}
```
